### DISCUSSION TITLE
WARNING: lib not found: api-ms-win-core-path-l1-1-0.dll dependency of c:\python39\python39.dll

### DISCUSSION THREAD

**Discussion by ghost** [pyinstaller/pyinstaller#6112] [Help] [answered]
`PS F:\CalculatorV2> pyinstaller --path "C:\Program Files (x86)\Windows Kits\10\Redist\10.0.19041.0\ucrt\DLLs\x64" calculator.py
210 INFO: PyInstaller: 4.3
211 INFO: Python: 3.9.6
231 INFO: Platform: Windows-10-10.0.19041-SP0
233 INFO: wrote F:\CalculatorV2\calculator.spec
240 INFO: UPX is not available.
252 INFO: Extending PYTHONPATH with paths
['F:\\CalculatorV2',
 'C:\\Program Files (x86)\\Windows '
 'Kits\\10\\Redist\\10.0.19041.0\\ucrt\\DLLs\\x64',
 'F:\\CalculatorV2']
pygame 2.0.1 (SDL 2.0.14, Python 3.9.6)
Hello from the pygame community. https://www.pygame.org/contribute.html
393 INFO: checking Analysis
393 INFO: Building Analysis because Analysis-00.toc is non existent
393 INFO: Initializing module dependency graph...
393 INFO: Caching module graph hooks...
424 WARNING: Several hooks defined for module 'win32ctypes.core'. Please take care they do not conflict.
439 INFO: Analyzing base_library.zip ...
5278 INFO: Processing pre-find module path hook distutils from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-distutils.py'.
5278 INFO: distutils: retargeting to non-venv dir 'c:\\python39\\lib'
11196 INFO: Caching module dependency graph...
11728 INFO: running Analysis Analysis-00.toc
11759 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\python39\python.exe
11931 WARNING: lib not found: api-ms-win-core-path-l1-1-0.dll dependency of c:\python39\python39.dll
12246 INFO: Analyzing F:\CalculatorV2\calculator.py
12246 INFO: Processing module hooks...
12246 INFO: Loading module hook 'hook-difflib.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12262 INFO: Loading module hook 'hook-distutils.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12262 INFO: Loading module hook 'hook-distutils.util.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12262 INFO: Loading module hook 'hook-encodings.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12496 INFO: Loading module hook 'hook-heapq.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12496 INFO: Loading module hook 'hook-lib2to3.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12809 INFO: Loading module hook 'hook-multiprocessing.util.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12809 INFO: Loading module hook 'hook-pickle.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12809 INFO: Loading module hook 'hook-sysconfig.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12809 INFO: Loading module hook 'hook-xml.etree.cElementTree.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12809 INFO: Loading module hook 'hook-xml.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
12949 INFO: Loading module hook 'hook-_tkinter.py' from 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks'...
13414 INFO: checking Tree
13415 INFO: Building Tree because Tree-00.toc is non existent
13415 INFO: Building Tree Tree-00.toc
13625 INFO: checking Tree
13625 INFO: Building Tree because Tree-01.toc is non existent
13625 INFO: Building Tree Tree-01.toc
13859 INFO: checking Tree
13859 INFO: Building Tree because Tree-02.toc is non existent
13859 INFO: Building Tree Tree-02.toc
13891 INFO: Looking for ctypes DLLs
13969 INFO: Analyzing run-time hooks ...
13984 INFO: Including run-time hook 'c:\\python39\\lib\\site-packages\\PyInstaller\\hooks\\rthooks\\pyi_rth_multiprocessing.py'
14000 INFO: Looking for dynamic libraries
14315 INFO: Looking for eggs
14315 INFO: Using Python library c:\python39\python39.dll
14315 INFO: Found binding redirects:
[]
14330 INFO: Warnings written to F:\CalculatorV2\build\calculator\warn-calculator.txt
14440 INFO: Graph cross-reference written to F:\CalculatorV2\build\calculator\xref-calculator.html
14471 INFO: checking PYZ
14471 INFO: Building PYZ because PYZ-00.toc is non existent
14471 INFO: Building PYZ (ZlibArchive) F:\CalculatorV2\build\calculator\PYZ-00.pyz
16005 INFO: Building PYZ (ZlibArchive) F:\CalculatorV2\build\calculator\PYZ-00.pyz completed successfully.
16037 INFO: checking PKG
16037 INFO: Building PKG because PKG-00.toc is non existent
16038 INFO: Building PKG (CArchive) PKG-00.pkg
16089 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
16093 INFO: Bootloader c:\python39\lib\site-packages\PyInstaller\bootloader\Windows-64bit\run.exe
16093 INFO: checking EXE
16095 INFO: Building EXE because EXE-00.toc is non existent
16096 INFO: Building EXE from EXE-00.toc
16101 INFO: Copying icons from ['c:\\python39\\lib\\site-packages\\PyInstaller\\bootloader\\images\\icon-console.ico']
16213 INFO: Writing RT_GROUP_ICON 0 resource with 104 bytes
16214 INFO: Writing RT_ICON 1 resource with 3752 bytes
16217 INFO: Writing RT_ICON 2 resource with 2216 bytes
16219 INFO: Writing RT_ICON 3 resource with 1384 bytes
16221 INFO: Writing RT_ICON 4 resource with 37019 bytes
16221 INFO: Writing RT_ICON 5 resource with 9640 bytes
16222 INFO: Writing RT_ICON 6 resource with 4264 bytes
16223 INFO: Writing RT_ICON 7 resource with 1128 bytes
16233 INFO: Appending archive to EXE F:\CalculatorV2\build\calculator\calculator.exe
17302 INFO: Building EXE from EXE-00.toc completed successfully.
17318 INFO: checking COLLECT
17318 INFO: Building COLLECT because COLLECT-00.toc is non existent
17318 INFO: Building COLLECT COLLECT-00.toc
17974 INFO: Building COLLECT COLLECT-00.toc completed successfully.
PS F:\CalculatorV2>`

I have already tried installing Windows SDK and Reinstalling Python.
The exe is created and does run on my pc without any problem.
I gave it to my friend with windows 7 (I have windows 10) and it showed this
![image](https://user-images.githubusercontent.com/88816736/129142144-7e1bf821-71c0-4f54-92dc-a2e21890c2d0.png)

Here is the program code:

`import time
run = True
while run:
    try:
        num1 = float(input("Enter a number: "))
        op = input("Enter a operation(+ , - , * , /)")
        num2 = float(input("Enter a number: "))
        result = 0

        if op == '+':
            result = num1 + num2
            print('.')
            time.sleep(1)
            print('..')
            time.sleep(1)
            print('...')
            time.sleep(1)
            print(result)

        elif op == '-':
            result = num1 - num2
            print('.')
            time.sleep(1)
            print('..')
            time.sleep(1)
            print('...')
            time.sleep(1)
            print(result)  

        elif op == '*':
            result = num1 * num2
            print('.')
            time.sleep(1)
            print('..')
            time.sleep(1)
            print('...')
            time.sleep(1)
            print(result)

        elif op == '/':
            result = num1 / num2
            print('.')
            time.sleep(1)
            print('..')
            time.sleep(1)
            print('...')
            time.sleep(1)
            print(result)
        else:
            print("That is not a valid operator!\n+ = Addition\n- = Subtraction\n* = Multiplication\n/ = Division")

    except ZeroDivisionError:
        print("You cannot divide by 0!!!")
    except ValueError:
        print("Please enter integers!!!")

    runAgain = input("Do you want to play again? (yes/no): ")
    if runAgain != "yes":
        break`

Its a really basic program.

Please help me out. Thanks.

**Comment by rokm** [comment, marked as answer]
Python 3.9 (and its DLLs, which PyInstaller-frozen application uses) is not compatible with Windows 7.

From [here](https://docs.python.org/3/using/windows.html#:%7E:text=As%20specified%20in%20PEP%2011,support%2C%20please%20install%20Python%203.8.):

```
As specified in PEP 11, a Python release only supports a Windows platform while Microsoft considers 
the platform under extended support. This means that Python 3.9 supports Windows 8.1 and newer. 
If you require Windows 7 support, please install Python 3.8.
```

**Reply by ghost** [reply]
Oh i get it now, i am sorry to waste your time.
